js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import screenfull from "screenfull";
+import butterchurn from "butterchurn";
+import reactionDiffusion2 from "butterchurn-presets/presets/converted/Geiss - Reaction Diffusion 2";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+  }
+  componentDidMount() {
+    const scale = window.devicePixelRatio;
+    const analyserNode = this.props.analyser;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width * scale,
+        height: this._canvasNode.height * scale
+      }
+    );
+    this.visualizer.connectAudio(analyserNode);
+    this.visualizer.loadPreset(reactionDiffusion2);
+    this._renderViz();
+
+    this._canvasNode.addEventListener("dblclick", () => {
+      this._handleRequestFullsceen();
+    });
```

Comment:
Optional nit: This could just be a `onDoubleClick` prop on the `<canvas>` React node.

Reply:
oh yea, definitely